docs/extensibility/extend-rest-api-update-cart.md

```diff
@@ -0,0 +1,90 @@
+use Automattic\WooCommerce\Blocks\Domain\Services\ExtendRestApi;
+
+add_action('woocommerce_blocks_loaded', function() {
+ // ExtendRestApi is stored in the container as a shared instance between the API and consumers.
+ // You shouldn't initiate your own ExtendRestApi instance using `new ExtendRestApi` but should always use the shared instance from the Package dependency injection container.
+ $extend = Package::container()->get( ExtendRestApi::class );
+ $extend->register_update_callback(
+   [
+    'namespace' => 'my-extensions-unique-namespace',
+    'callback'  => /* Add your callable here */
+   ]
+ )
+} );
+```
+
+The method takes a single argument, an associative array.
+
+The associative array must have the following entries:
+- `namespace` (string) - The unique namespace of your extension.
+- `callback` (Callable) - The function/method (or Callable) that will be executed when the `cart/extensions` endpoint is
+  hit with a `namespace` that matches your extension's. The callable should take a single argument. The data passed into
+  the callback via this argument will be an array containing whatever data you choose to pass to it.
+More information on how data gets passed from the client to the callback will be detailed over the next sections.
+
+The callable does not need to return anything, if it does, then its return value will not be used.
+
+## Making the request to the `cart/extensions` endpoint
+When you wish to make a request to the endpoint, you should do it by calling the `extensionCartUpdate` function, available
+from `@woocommerce/blocks-checkout`.
+
+This function takes a single argument, of type object. The object must contain two properties: `namespace` and `data`.
+
+- `namespace` should be the same as the one you used when registering your callback with `ExtendRestApi` on the
+server-side.
+- `data` is an object containing the data you want to pass to the server-side function. These data must be serializable.
+
+```javascript
+import { extensionCartUpdate } from '@woocommerce/blocks-checkout';
+
+extensionCartUpdate(
+  {
+    namespace: 'my-extensions-unique-namespace',
+    data: {
+      key: 'value',
+      another_key: 100,
+      third_key: {
+        fourth_key: true,
+      }
+    }
+  }
+);
+```
+
+When this function is executed, after a short delay (due to batch requests) a POST request will be made to the
+`cart/extensions` endpoint, the POSTed data will be the value of the `data` key in the argument object.
+
+When the request completes, the API will return the current server-side cart to the client. The client will then update
+the UI.
+
+## Why can't I just POST to `cart/extensions` directly?
+This endpoint returns an object representing the customer's cart, WooCommerce Blocks will then update the client-side
+stores with the new cart data. The methods to do this are not available to third party extensions, which is why you must
+use the `extensionCartUpdate` method.
```

Comment:
This can be part of the "Problem" section.
It also doesn't do a good job of explaining why you can't update Cart block yourself, which is to protect the block from getting broken by a malfunctioning plugin.

Reply:
Fixed by the new format, and also more detail about _why_ they can't update it has been added to the Problem section, and also mentioned again in the Things to consider section.